pandas/core/arrays/period.py

```diff
@@ -277,7 +278,7 @@ def _check_compatible_with(self, other):
     def dtype(self):
         return self._dtype
 
-    @property
+    @property  # type: ignore # read-only property overwriting read/write
```

Comment:
Not really sure what to do with this. `Period` defines this property without a setter, but its superclass `DateTimeLikeArrayMixin` defines freq as a read-write property.

Reply:
Seems like this could be a legitimate bug. Assigning to `freq` in the superclass would impact the getter (setting and returning `self._freq`) but assignment in the subclass would not really do anything, since the getter returns `self.dtype.freq`.

@jbrockmendel any thoughts here?

Reply:
On further investigation I suppose this isn't a bug as trying to assign `freq` on a PeriodArray will still raise an error, but this type of design is potentially discouraged due to the fact that PeriodArray no longer fully supports all of the methods that DateTimeLikeArrayMixin does (namely the assignment).

I think ignoring for now is best approach to not go too far down a rabbit hole, but will defer if anyone else objects

Reply:
can you put the comments aboe